test/e2e/specs/preview.test.js

```diff
@@ -6,23 +6,36 @@ import { parse } from 'url';
 /**
  * Internal dependencies
  */
-import '../support/bootstrap';
 import {
 	newPost,
-	newDesktopBrowserPage,
 	getUrl,
 	publishPost,
 } from '../support/utils';
 
 describe( 'Preview', () => {
-	beforeAll( async () => {
-		await newDesktopBrowserPage();
-	} );
-
 	beforeEach( async () => {
 		await newPost();
 	} );
 
+	async function openPreviewPage( editorPage ) {
+		let openTabs = await browser.pages();
+		expect( openTabs ).toHaveLength( 2 );
+		await editorPage.click( '.editor-post-preview' );
+
+		// Wait for the new tab to open.
+		while ( 3 !== openTabs.length ) {
+			await editorPage.waitFor( 250 );
```

Comment:
I thought we might have had a lint rule forbidding this. Timeouts are generally discouraged. In this case, it's similar in purpose to [`waitForFunction`](https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#pagewaitforfunctionpagefunction-options-args) which is effectively implemented by polling by default, though using `requestAnimationFrame`. Similarly, I'd like to see a much smaller interval here, even as low as `10`, `1`, or `0`. `250` is too long.

Reply:
`waitForFunction` doesn't seem to be a good fit because it is evaluated in browser context.

Reply:
@aduth With `targetcreated` not being reliable, and the cause of most of the intermittent failures I've run up against in travis - the "Execution context lost" failure message was caused by that. So I don't see any other alternative than a timeout. Happy to change it to 10 though.

https://github.com/WordPress/gutenberg/pull/8301#discussion_r206726817 is where it was agreed to use `waitFor`. But I'm happy to change it back to a custom `sleep` function again if that's preferred.